test/unit/BCoWPool.t.sol

```diff
@@ -124,11 +124,33 @@ contract BCoWPool_Unit_Commit is BaseCoWPoolTest {
     bCoWPool.commit(orderHash);
   }
 
+  function test_Revert_CommitmentAlreadySet(bytes32 _existingCommitment, bytes32 _newCommitment) public {
+    vm.assume(_existingCommitment != bytes32(0));
+    bCoWPool.call__setLock(_existingCommitment);
+    vm.prank(cowSolutionSettler);
+    vm.expectRevert(IBCoWPool.BCoWPool_CommitmentAlreadySet.selector);
+    bCoWPool.commit(_newCommitment);
+  }
+
   function test_Set_Commitment(bytes32 orderHash) public {
     vm.prank(cowSolutionSettler);
     bCoWPool.commit(orderHash);
     assertEq(bCoWPool.commitment(), orderHash);
   }
+
+  function test_Call_SetLock(bytes32 orderHash) public {
+    bCoWPool.expectCall__setLock(orderHash);
+    vm.prank(cowSolutionSettler);
+    bCoWPool.commit(orderHash);
+  }
+
+  function test_Set_ReentrancyLock(bytes32 orderHash) public {
+    vm.assume(orderHash != _MUTEX_FREE);
+    vm.prank(cowSolutionSettler);
+    bCoWPool.commit(orderHash);
+    assertNotEq(bCoWPool.call__getLock(), _MUTEX_FREE);
```

Comment:
is this needed? :0

Reply:
Not for the test itself, but I wanted to be explicit about it not ever being the value for a free mutex